x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -353,18 +353,25 @@ async function endpointListMiddleware({
 }) {
   const { getState, dispatch } = store;
 
-  const { page_index: pageIndex, page_size: pageSize } = uiQueryParams(getState());
+  const {
+    page_index: pageIndex,
+    page_size: pageSize,
+    sort_field: sortField,
+    sort_direction: sortDirection,
+  } = uiQueryParams(getState());
   let endpointResponse: MetadataListResponse | undefined;
 
   try {
     const decodedQuery: Query = searchBarQuery(getState());
+    const sorting = sortField ? { sortField, sortDirection } : {};
```

Comment:
it turned out that it's perfectly fine, because `coreStart.http.get` filters undefined values from the request. so I simply removed this whole check
3bb8616ae3226cfc34b0a2e9313fa6b7d1f40ce3